storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/MetricCollector.java

```diff
@@ -55,41 +58,67 @@ class MetricCollector implements MetricPublisher {
+        requestMetrics.put("GetObject", createRequestsSensor("get-object-requests"));
+        latencyMetrics.put("GetObject", createLatencySensor("get-object-time"));
+        requestMetrics.put("UploadPart", createRequestsSensor("upload-part-requests"));
+        latencyMetrics.put("UploadPart", createLatencySensor("upload-part-time"));
+        requestMetrics.put("CreateMultipartUpload", createRequestsSensor("create-multipart-upload-requests"));
+        latencyMetrics.put("CreateMultipartUpload", createLatencySensor("create-multipart-upload-time"));
+        requestMetrics.put("CompleteMultipartUpload", createRequestsSensor("complete-multipart-upload-requests"));
+        latencyMetrics.put("CompleteMultipartUpload", createLatencySensor("complete-multipart-upload-time"));
+        requestMetrics.put("PutObject", createRequestsSensor("put-object-requests"));
+        latencyMetrics.put("PutObject", createLatencySensor("put-object-time"));
+        requestMetrics.put("DeleteObject", createRequestsSensor("delete-object-requests"));
+        latencyMetrics.put("DeleteObject", createLatencySensor("delete-object-time"));
+        requestMetrics.put("AbortMultipartUpload", createRequestsSensor("abort-multipart-upload-requests"));
+        latencyMetrics.put("AbortMultipartUpload", createLatencySensor("abort-multipart-upload-time"));
+
+        errorMetrics.put(THROTTLING.toString(), createRequestsSensor("throttling-errors"));
+        errorMetrics.put(SERVER_ERROR.toString(), createRequestsSensor("server-errors"));
+        errorMetrics.put(CONFIGURED_TIMEOUT.toString(), createRequestsSensor("configured-timeout-errors"));
+        errorMetrics.put(IO.toString(), createRequestsSensor("io-errors"));
+        errorMetrics.put(OTHER.toString(), createRequestsSensor("other-errors"));
     }
 
-    private Sensor createSensor(final String name) {
+    private Sensor createRequestsSensor(final String name) {
         final Sensor sensor = metrics.sensor(name);
         sensor.add(metrics.metricName(name + "-rate", METRIC_GROUP), new Rate());
         sensor.add(metrics.metricName(name + "-total", METRIC_GROUP), new CumulativeCount());
         return sensor;
     }
 
+    private Sensor createLatencySensor(final String name) {
+        final Sensor sensor = metrics.sensor(name);
+        sensor.add(metrics.metricName(name + "-max", METRIC_GROUP), new Max());
+        sensor.add(metrics.metricName(name + "-avg", METRIC_GROUP), new Avg());
+        return sensor;
+    }
+
     @Override
     public void publish(final MetricCollection metricCollection) {
         final List<String> metricValues = metricCollection.metricValues(CoreMetric.OPERATION_NAME);
-        for (final String metricValue : metricValues) {
-            final var sensor = requestMetrics.get(metricValue);
-            if (sensor != null) {
-                sensor.record();
+        // metrics are reported per request, so 1 value can be assumed.
+        if (metricValues.size() == 1) {
```

Comment:
Hm, I'm thinking that maybe it's worth logging a warning if the size is different from 1

Reply:
I was thinking the same actually, so yeah, lets have it.

Reply:
Make sense. Adding it now.